src/components/notification/NotificationHelpers.ts

```diff
@@ -0,0 +1,39 @@
+import { AssessmentCategories } from '../assessment/assessmentShape';
+import { Notification } from './notificationShape';
+
+type filterByIdOptions = {
+  assessment_id?: number;
+  submission_id?: number;
+};
+
+type filterByTypeOptions = AssessmentCategories | 'Grading';
+
+export function filterNotificationsById(
```

Comment:
Actually instead of going with this its simply cleaner to split into two functions which have a clear purpose:

`filterNotificationsByAssessment` and `filterNotificationsByAssessmentBySubmission`